src/taipy/core/notification/event.py

```diff
@@ -71,30 +74,60 @@ class Event:
             and `SUBMISSION`) that was performed on the entity.
         attribute_name (Optional[str]): Name of the entity's attribute changed. Only relevant for `UPDATE`
             operations
+        attribute_value (Optional[str]): Name of the entity's attribute changed. Only relevant for `UPDATE`
+            operations
+        metadata (dict): A dict of additional medata about the source of this event
         creation_date (datetime): Date and time of the event creation.
     """
 
-    def __init__(
-        self,
-        entity_type: EventEntityType,
-        entity_id: Optional[str],
-        operation: EventOperation,
-        attribute_name: Optional[str] = None,
-    ):
-        self.creation_date = datetime.now()
-        self.entity_id = entity_id
-        self.entity_type = entity_type
-        self.operation = self.__preprocess_operation(operation, entity_type)
-        self.attribute_name = self.__preprocess_attribute_name(attribute_name, operation)
-
-    @classmethod
-    def __preprocess_attribute_name(cls, attribute_name: Optional[str], operation: EventOperation) -> Optional[str]:
-        if operation in _NO_ATTRIBUTE_NAME_OPERATIONS and attribute_name is not None:
-            raise InvalidEventAttributeName
-        return attribute_name
+    entity_type: EventEntityType
+    operation: EventOperation
+    entity_id: Optional[str] = None
+    config_id: Optional[str] = None
+    attribute_name: Optional[str] = None
+    attribute_value: Optional[Any] = None
+
+    metadata: dict = field(default_factory=dict)
+    creation_date: datetime = field(init=False)
+
+    def __post_init__(self):
+        # Creation date
+        super().__setattr__("creation_date", datetime.now())
 
-    @classmethod
-    def __preprocess_operation(cls, operation: EventOperation, entity_type: EventEntityType) -> EventOperation:
-        if entity_type in _UNSUBMITTABLE_ENTITY_TYPES and operation == EventOperation.SUBMISSION:
+        # Check operation:
+        if self.entity_type in _UNSUBMITTABLE_ENTITY_TYPES and self.operation == EventOperation.SUBMISSION:
             raise InvalidEventOperation
-        return operation
+
+        # Check attribute name:
+        if self.operation in _NO_ATTRIBUTE_NAME_OPERATIONS and self.attribute_name is not None:
+            raise InvalidEventAttributeName
+
+
+@singledispatch
+def make_event(
+    entity: Any,
+    operation: EventOperation,
+    /,
```

Comment:
I don't know this `/,` param. Is it a way to split mandatory positional parameters and optional parameters for readability? Is it standard?

Reply:
Actually, your remark makes me think the `*` might have been more appropriate in this case... 🤔 

```
def f(pos1, pos2, /, pos_or_kwd, *, kwd1, kwd2):
      -----------    ----------     ----------
        |             |                  |
        |        Positional or keyword   |
        |                                - Keyword only
         -- Positional only
```

Reply:
hmmm interesting, do you think it make sense for us to do this across all other functions?

Reply:
Honestly, I can't say. This would depend on the case I guess. 
But beware that applying this to existing functions would break API compatibility.
Let's discuss this offline.